src/ducks/apps/components/Sections/Sections.jsx

```diff
@@ -27,53 +27,51 @@ import Filters from './components/Filters'
 const Sections = ({ apps, error, onAppClick, filter, onFilterChange }) => {
   const { lang } = useI18n()
 
+  const previousFilter = usePreviousValue(filter)
   const [internalFilter, setInternalFilter] = useState({})
   const [searchFieldValue, setSearchFieldValue] = useState('')
-  const [searchResults, setSearchResults] = useState(null)
 
-  const options = {
-    includeScore: true,
-    includeMatches: true,
-    minMatchCharLength: 3,
-    threshold: 0.2,
-    ignoreLocation: true,
-    keys: [
-      { name: 'name', weight: 3 },
-      'categories',
-      `locales.${lang}.short_description`,
-      `locales.${lang}.long_description`
-    ]
-  }
+  const options = useMemo(
+    () => ({
+      includeScore: true,
+      includeMatches: true,
+      minMatchCharLength: 3,
+      threshold: 0.2,
+      ignoreLocation: true,
+      keys: [
+        { name: 'name', weight: 3 },
+        'categories',
+        `locales.${lang}.short_description`,
+        `locales.${lang}.long_description`
+      ]
+    }),
+    [lang]
+  )
 
-  const appsForSearch = useMemo(() => {
-    const filterMatcher = filterUtils.makeMatcherFromSearch(
-      filter || internalFilter
-    )
-    return apps.filter(filterMatcher).map(app => ({
+  const searchResults = useMemo(() => {
+    const filterMatcher = filterUtils.makeMatcherFromSearch(filter)
+    const filteredApps = apps.filter(filterMatcher).map(app => ({
       ...app,
       doctypes: app.permissions
         ? Object.values(app.permissions).map(x => x.type)
         : null
     }))
-  }, [apps, filter, internalFilter])
-
-  const onSearchChanged = useMemo(
-    () =>
-      debounce(value => {
-        const fuse = new Fuse(appsForSearch, options)
-        setSearchResults(value ? fuse.search(value) : null)
-      }, 300),
-    [appsForSearch, options]
-  )
+    const fuse = new Fuse(filteredApps, options)
```

Comment:
I know that this is not part of this PR, but this is weird, no? We should instanciate only once the `Fuse` object with all the apps, and then make a search on the filter and on the searchValue. We should not create this `Fuse` object every time, it doesn't make sense and should consume a lot of RAM.

Reply:
Yes it's weird now you have point out. There is an other thing that disturb me, the filters are applied twice once before the search and once app list. In the same refacto, maybe we can use fuse.js for the two.